fix/show-toast-error-in-collection-rating-guest

Mostafa-khaled21 opened the pull request:
https://github.com/user-attachments/assets/05592911-6f77-43e5-a995-b60a594c34ef

Mostafa-khaled21 committed:
Make EncryptionService cipher a local val

The cipher object was being reused which caused errors when multiple calls were made to the encrypt/decrypt functions. This change makes the cipher a local val so a new instance is created for each call.

Mostafa-khaled21 committed:
Refactor MyCollectionsViewModel to check login status before fetching collections

Mostafa-khaled21 committed:
Refactor: Check login status before fetching ratings in RatingViewModel

This commit modifies the `RatingViewModel` to check if a user is logged in before attempting to fetch their ratings and associated genres.

- Added `isLoggedInUseCase` to the ViewModel's dependencies.
- Introduced a `checkLoginStatus` function that is called in the `init` block.
- If the user is logged in, `getGenres()` is called to proceed with fetching data.
- If the user is not logged in, the loading state is set to `false` and no data fetching occurs.
- The `RatingScreenState` is updated with an `isLoggedIn` boolean flag.

mohamedebrahem13 commented on presentation/profile/src/main/java/com/giraffe/presentation/profile/screens/ratings/RatingViewModel.kt:
```diff
@@ -23,12 +24,30 @@ class RatingViewModel @Inject constructor(
+    }
+
+    private fun checkLoginStatus() {
+        safeExecute(
```
private fun checkLoginStatus() {
    safeExecute(
        onSuccess = ::handleLoginSuccess,
        onError = ::onFailure
    ) {
        isLoggedInUseCase()
    }
}

private fun handleLoginSuccess(loggedIn: Boolean) {
    updateState { it.copy(isLoggedIn = loggedIn) }

    if (loggedIn) {
        getCollections()
    } else {
        updateState { it.copy(isLoading = false) }
    }
}

Mostafa-khaled21 committed:
Refactor login status checking in RatingViewModel and MyCollectionsViewModel

This commit refactors the `checkLoginStatus` function in both `RatingViewModel.kt` and `MyCollectionsViewModel.kt`. The logic for handling a successful login check has been extracted into a new private function `handleLoginSuccess` in each ViewModel. This improves code readability and maintainability by separating concerns.
